tests/desktop/developer_hub/test_submit_fxos_add_on.py

```diff
@@ -0,0 +1,65 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import os
+import shutil
+import uuid
+from zipfile import ZipFile
+
+import pytest
+
+from pages.desktop.developer_hub.content_tools import ContentTools
+from tests.base_test import BaseTest
+
+
+class TestSubmitFxOSAddOn(BaseTest):
+
+    @pytest.fixture
+    def fxos_add_on_name(self, request, mozwebqa):
+
+        add_on_name = str(uuid.uuid4())
+
+        # backup the file we're changing, just in case!
+        shutil.copyfile(self._get_resource_path('fxaddon/manifest.json'), self._get_resource_path('manifest.bak'))
```

Comment:
I don't think we should be modifying the file in place. Let's copy to a temporary directory. I see that ZipFile has a [writestr](https://docs.python.org/2/library/zipfile.html#zipfile.ZipFile.writestr) method that should allow us to write a string to an existing zip file as a file. We should therefore be able to read the `manifest.json` directly from the zip file, set a unique name, and then `writestr` the replacement JSON contents.